fe2-android/app/src/main/java/com/github/dedis/popstellar/model/network/method/message/data/federation/FederationResult.kt

```diff
@@ -0,0 +1,84 @@
+package com.github.dedis.popstellar.model.network.method.message.data.federation
+
+import com.github.dedis.popstellar.model.network.method.message.MessageGeneral
+import com.github.dedis.popstellar.model.network.method.message.data.Action
+import com.github.dedis.popstellar.model.network.method.message.data.Data
+import com.github.dedis.popstellar.model.network.method.message.data.Objects
+import com.github.dedis.popstellar.utility.MessageValidator
+import com.google.gson.annotations.SerializedName
+
+/** Informs about the result of the authentication procedure */
+class FederationResult
+/**
+ * Constructor for a data Federation Result
+ *
+ * @param status status of the result (either success or failure)
+ * @param reason reason of the failure
+ * @param publicKey public key of the other LAO organizer
+ * @param challenge challenge used to connect the LAOs
+ */
+(
+    val status: String,
+    val reason: String? = null,
+    @SerializedName("public_key") val publicKey: String? = null,
+    val challenge: MessageGeneral,
+) : Data {
+
+  init {
+    when (status) {
```

Comment:
This looks like input validation, maybe you should complete MessageValidator class to check a federation status? Validation would then just be a clear oneliner 👍